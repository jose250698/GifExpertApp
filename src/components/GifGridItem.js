import React from 'react';
import PropTypes from 'prop-types';


const GifGridItem = ({id, title, url}) => {
    return (
        <div className="card animate__animated animate__fadeIn">
            <img id={id} src={url} alt={title}/>
            <p>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    id: PropTypes.oneOfType([
        PropTypes.number,
        PropTypes.string
    ]).isRequired,
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
};

export default GifGridItem;