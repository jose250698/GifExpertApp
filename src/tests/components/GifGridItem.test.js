import {shallow} from 'enzyme';
import React from 'react';
import GifGridItem from '../../components/GifGridItem';

describe('<GifGridItem /> tests', () => {
    const id = 354;
    const title = 'A title';
    const url = 'http://localhost/some_image.jpg';
    const wrapper = shallow(<GifGridItem id={id} title={title} url={url} />);

    test('should show component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('should have a p element with the title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    });

    test('should have an image with the same props', () => {
        const img = wrapper.find('img');
        expect(img.prop('id')).toBe(id);
        expect(img.prop('alt')).toBe(title);
        expect(img.prop('src')).toBe(url);
    });

    test('should have the animate__fadeIn class', () => {
        const div = wrapper.find('div');
        console.log(div.props());
        expect(div.hasClass('animate__fadeIn')).toBe(true);
    });
})
