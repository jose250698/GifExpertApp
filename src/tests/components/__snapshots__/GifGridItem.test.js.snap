// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGridItem /> tests should show component correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="A title"
    id={354}
    src="http://localhost/some_image.jpg"
  />
  <p>
    A title
  </p>
</div>
`;
